packages/cactus-plugin-ledger-connector-ethereum/src/main/typescript/plugin-ledger-connector-ethereum.ts

```diff
@@ -943,6 +943,22 @@ export class PluginLedgerConnectorEthereum
   ): Promise<RunTransactionResponse> {
     Checks.truthy(req, "deployContract() request arg");
 
+    // Validate the keys in the request object
+    const validKeys = [
+      "web3SigningCredential",
+      "contract",
+      "constructorArgs",
+      "gasConfig",
+      "value",
+    ];
+    const extraKeys = Object.keys(req).filter(
+      (key) => !validKeys.includes(key),
+    );
+
+    if (extraKeys.length > 0) {
+      throw new Error(`Invalid parameters: ${extraKeys.join(", ")}`);
+    }
+
```

Comment:
@ashnashahgrover The Open API spec is already declaring the request object with additional properties being forbidden, so in theory the openapi validator should reject any request with random other properties in them even without this change.

Could you please confirm if this logic is working (without the code you added here) and if it's not then please also open a separate issue to fix the root cause in the open API validation logic.

The relevant part of the spec I'm talking about:

`"additionalProperties": false,`

```json
"DeployContractV1Request": {
        "type": "object",
        "required": ["web3SigningCredential", "contract"],
        "additionalProperties": false,
        "properties": {
          "web3SigningCredential": {
            "$ref": "#/components/schemas/Web3SigningCredential"
          },
          "contract": {
            "oneOf": [
              {
                "$ref": "#/components/schemas/ContractJsonDefinition"
              },
              {
                "$ref": "#/components/schemas/ContractKeychainDefinition"
              }
            ],
            "nullable": false
          },
          "constructorArgs": {
            "description": "The list of arguments to pass in to the constructor of the contract being deployed.",
            "type": "array",
            "default": [],
            "items": {}
          },
          "gasConfig": {
            "$ref": "#/components/schemas/GasTransactionConfig"
          },
          "value": {
            "type": "string",
            "description": "Ether balance to send on deployment.",
            "nullable": false
          }
        }
      },
```

Reply:
Yes, I had the same query @petermetz , to which @ashnashahgrover has mentioned that even with additional properties, it still pass, which was very strange.
@ashnashahgrover can you confirm on the above once again?

Reply:
@jagpreetsinghsasan @ashnashahgrover OK, that sounds unfortunately like a completely different bug in the validation itself to be fixed. Could you please open a bug with **detailed** reproduction steps and then we can work on fixing that separately.

Reply:
Yes even with additional properties it fails, and I am investigating why. Will open a seperate bug for this!